src/main/java/com/technicjelle/Cheese.java

```diff
@@ -173,20 +174,24 @@ public static Cheese createFromCellsUnsafe(Vector2d cellSize, Vector2i... cells)
 	 * @see #createMultiCheeseFromChunks(Vector2i...)
 	 */
 	public static boolean checkConnected(Vector2i... cells) {
-		Set<Vector2i> cellsToCheck = new HashSet<>(List.of(cells));
+		if (cells.length == 0) return false;
+		if (cells.length == 1) return true;
+
+		Set<Vector2i> cellsToCheck = Set.of(cells);
 		Set<Vector2i> visited = new HashSet<>();
-		Queue<Vector2i> toVisit = new LinkedList<>();
-		toVisit.add(cellsToCheck.iterator().next()); // start with the first cell
+		Stack<Vector2i> toVisit = new Stack<>();
+		toVisit.push(cells[0]); // start with the first cell
+		visited.add(cells[0]);
 
 		while (!toVisit.isEmpty()) {
-			Vector2i current = toVisit.poll();
-			visited.add(current);
+			Vector2i current = toVisit.pop();
 
 			// add all neighbours that are in the cellsToCheck set and not visited yet
 			for (Direction direction : Direction.values()) {
 				Vector2i neighbour = current.add(direction.vector);
 				if (cellsToCheck.contains(neighbour) && !visited.contains(neighbour)) {
```

Comment:
you do add and contains. Instead I suggest to do: 

`if(cellsToCheck.contains(neighbour) && !visited.add(neighbour))`
since actually adding the neighbour is an optional operation, you might get a few cycles out of re-using the return value of the add check! :)

Reply:
This does seem to pass the unit tests, yeah :eyes:
I'm benchmarking it right now, to see if it's any faster!
```java
// add all neighbours that are in the cellsToCheck set and not visited yet
for (Direction direction : Direction.values()) {
	Vector2i neighbour = current.add(direction.vector);
	if (cellsToCheck.contains(neighbour) && visited.add(neighbour)) {
		toVisit.push(neighbour);
	}
}
```

Reply:
```
Benchmark                          Mode  Cnt    Score   Error  Units
CheeseBench.measureCheese10        avgt        50.807          us/op
CheeseBench.measureCheese11        avgt        63.525          us/op
CheeseBench.measureCheese12        avgt        74.523          us/op
CheeseBench.measureCheese13        avgt        88.670          us/op
CheeseBench.measureCheese14        avgt       121.484          us/op
CheeseBench.measureCheese15        avgt       135.255          us/op
CheeseBench.measureCheese16        avgt       150.883          us/op
```
```
Benchmark                          Mode  Cnt    Score   Error  Units
CheeseBench.measureCheese10        avgt        49.861          us/op
CheeseBench.measureCheese11        avgt        63.353          us/op
CheeseBench.measureCheese12        avgt        74.497          us/op
CheeseBench.measureCheese13        avgt        85.232          us/op
CheeseBench.measureCheese14        avgt       117.236          us/op
CheeseBench.measureCheese15        avgt       130.292          us/op
CheeseBench.measureCheese16        avgt       142.694          us/op
```
The second one is a bit faster, isn't it?
Very nice! Thanks!

Reply:
Implemented in 25365437c3f8712fbd972345f4c89bf9fc110446! Thanks! :D